### PR TITLE
#327 Improve notifcation

### PR DESCRIPTION
**Pull request by 3rdvision**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-components-vue/issues/327 |
| Decisions | Improve notification appearance. |
| Animated GIF | uiiiii |

### CODE REVIEW

File: vue/components/ui/molecules/notification/notification.vue [hunk 1]
```diff
@@ -32,36 +38,45 @@
...
 }
 
+.notification.dark > .notification-container {
+    background-color: $dark;
+    color: $white;
+}
+
 .notification > .notification-container > .icon {
-    padding-right: 8px;
+    display: flex;
```
Comment (gcandal) [inline vue/components/ui/molecules/notification/notification.vue:78]: Why is this needed?

File: vue/components/ui/molecules/notification/notification.vue [hunk 2]
```diff
@@ -32,36 +38,45 @@
...
-    border-radius: 6px 6px 6px 6px;
-    box-shadow: 0px 0px 16px rgba(45, 58, 70, 0.25);
+    border-radius: 8px 8px 8px 8px;
+    box-shadow: 0 3px 5px -1px #00000033, 0 6px 10px 0 #00000024, 0 1px 18px 0 #0000001f;
     box-sizing: border-box;
     color: #1d2631;
     display: inline-flex;
-    font-size: 13px;
-    font-weight: 600;
+    font-family: "Roboto Slab", MarkPro, sans-serif;
```
Comment (gcandal) [inline vue/components/ui/molecules/notification/notification.vue:56]: Let's keep the font we have.

File: css/animations.scss [hunk 3]
```diff
@@ -227,12 +227,12 @@
         -khtml-opacity: 0;
         -webkit-opacity: 0;
         opacity: 0;
-        -o-transform: scale(0.96);
-        -ms-transform: scale(0.96);
-        -moz-transform: scale(0.96);
-        -khtml-transform: scale(0.96);
-        -webkit-transform: scale(0.96);
-        transform: scale(0.96);
+        -o-transform: scale(0.85);
```
Comment (gcandal) [inline css/animations.scss:230]: Re-define this animation in the component itself, so we don't change it for every component.

File: vue/components/ui/molecules/notification/notification.vue [hunk 4]
```diff
@@ -32,36 +38,45 @@
...
+    font-size: 14px;
+    font-weight: 500;
     justify-content: center;
     letter-spacing: 0.3px;
-    line-height: 18px;
+    line-height: 24px;
+    line-height: 24px;
```
Comment (gcandal) [inline vue/components/ui/molecules/notification/notification.vue:62]: Duplicated.

File: vue/components/ui/molecules/notification/notification.vue [hunk 5]
```diff
@@ -32,36 +38,45 @@
     text-align: center;
     top: 100px;
     z-index: 50;
+    transition: opacity 0.125s ease-in-out;
```
Comment (gcandal) [inline vue/components/ui/molecules/notification/notification.vue:41]: Do we need both transition and animation?

File: vue/components/ui/molecules/notification/notification.vue (hunk 1 above)
Comment (3rdvision) [reply vue/components/ui/molecules/notification/notification.vue:78]: This is neede in a specific situation in which the text container will be taller than the notification icon.
With this:
image

Without this:
image

File: vue/components/ui/molecules/notification/notification.vue (hunk 5 above)
Comment (3rdvision) [reply vue/components/ui/molecules/notification/notification.vue:41]: Yes, the animation is for element-enter and the transition (opacity) is for element-leave

File: vue/components/ui/molecules/notification/notification.vue (hunk 1 above)
Comment (gcandal) [reply vue/components/ui/molecules/notification/notification.vue:77]: Can't you solve that with `vertical-align`?

Comment (3rdvision) [reply vue/components/ui/molecules/notification/notification.vue:77]: Not possible as discussed

File: vue/components/ui/molecules/notification/notification.vue [hunk 6]
```diff
@@ -31,37 +37,112 @@
...
+    color: $white;
+    display: flex;
+}
+
 .notification > .notification-container > .icon {
-    padding-right: 8px;
+    margin-right: 16px;
+}
+
+@keyframes fade-grow {
```
Comment (joamag) [inline vue/components/ui/molecules/notification/notification.vue:82]: rename to `fade-grow-smaller` and move to animations.scss

File: vue/components/ui/molecules/notification/notification.vue [hunk 7]
```diff
@@ -31,37 +37,112 @@
...
-    border-radius: 6px 6px 6px 6px;
-    box-shadow: 0px 0px 16px rgba(45, 58, 70, 0.25);
+    border-radius: 8px 8px 8px 8px;
+    box-shadow: 0 3px 5px -1px #00000033, 0 6px 10px 0 #00000024, 0 1px 18px 0 #0000001f;
     box-sizing: border-box;
     color: #1d2631;
     display: inline-flex;
-    font-size: 13px;
-    font-weight: 600;
+    font-family: "Roboto Slab", MarkPro, sans-serif;
```
Comment (joamag) [inline vue/components/ui/molecules/notification/notification.vue:57]: remove this font

File: vue/components/ui/molecules/notification/notification.vue [hunk 8]
```diff
@@ -31,37 +37,112 @@
...
 
 .notification > .notification-container {
-    animation: fade-grow-rise 0.35s cubic-bezier(0.645, 0.045, 0.355, 1);
+    align-items: center;
+    animation: fade-grow 0.125s cubic-bezier(0.645, 0.045, 0.355, 1);
     background-color: $white;
-    border-radius: 6px 6px 6px 6px;
-    box-shadow: 0px 0px 16px rgba(45, 58, 70, 0.25);
+    border-radius: 8px 8px 8px 8px;
+    box-shadow: 0 3px 5px -1px #00000033, 0 6px 10px 0 #00000024, 0 1px 18px 0 #0000001f;
```
Comment (joamag) [inline vue/components/ui/molecules/notification/notification.vue:53]: we'll need to have just a single color

**Comment by joamag** [comment]
@gcandal please to check these topics so that we can merge this one easily
